cpp/include/cuml/decomposition/params.hpp

```diff
@@ -31,24 +31,24 @@ enum class solver : int {
 
 class params {
  public:
-  int n_rows;
-  int n_cols;
+  std::size_t n_rows;
+  std::size_t n_cols;
   int gpu_id = 0;
 };
 
 class paramsSolver : public params {
  public:
   // math_t tol = 0.0;
-  float tol        = 0.0;
-  int n_iterations = 15;
-  int verbose      = 0;
+  float tol                  = 0.0;
+  std::uint32_t n_iterations = 15;
+  int verbose                = 0;
 };
 
 template <typename enum_solver = solver>
 class paramsTSVDTemplate : public paramsSolver {
  public:
-  int n_components      = 1;
-  enum_solver algorithm = enum_solver::COV_EIG_DQ;
+  std::size_t n_components = 1;
```

Comment:
In theory, `n_components` could be as large as n_cols (or n_rows, whichever is smaller) but in practice this should never grow beyond 32-bits. However, if we are going to keep n_rows and n_cols as 64-bits, I'm okay keeping this as 64 as well.